GitUI/UserControls/BlameControl.cs

```diff
@@ -188,99 +219,59 @@ private void BlameCommitter_ScrollPosChanged(object sender, EventArgs e)
+            _changingScrollPosition = false;
         }
 
-        private readonly AsyncLoader _blameLoader = new AsyncLoader();
-
-        public void LoadBlame(GitRevision revision, List<string> children, string fileName, RevisionGrid revGrid, Control controlToMask, Encoding encoding, int? initialLine = null, bool force = false)
+        private void ProcessBlame(GitRevision revision, List<string> children, Control controlToMask, int lineNumber, int scrollpos)
         {
-            string guid = revision.Guid;
-
-            // refresh only when something changed
-            if (!force && guid == _blameHash && fileName == _fileName && revGrid == _revGrid && encoding == _encoding)
-            {
-                return;
-            }
-
-            controlToMask?.Mask();
-
-            var scrollpos = BlameFile.ScrollPos;
+            var gutter = new StringBuilder(capacity: 4096);
+            var body = new StringBuilder(capacity: 4096);
 
-            int line = initialLine.GetValueOrDefault(0);
-            if (_clickedBlameLine.CommitGuid == guid)
-            {
-                line = _clickedBlameLine.OriginLineNumber;
-            }
-
-            _revGrid = revGrid;
-            _fileName = fileName;
-            _encoding = encoding;
+            GitBlameCommit lastCommit = null;
 
-            _blameLoader.Load(() => _blame = Module.Blame(fileName, guid, encoding),
-                () => ProcessBlame(revision, children, controlToMask, line, scrollpos));
-        }
-
-        private void ProcessBlame(GitRevision revision, List<string> children, Control controlToMask, int line, int scrollpos)
-        {
-            var blameCommitter = new StringBuilder();
-            var blameFile = new StringBuilder();
-            for (int i = 0; i < _blame.Lines.Count; i++)
+            foreach (var line in _blame.Lines)
             {
-                GitBlameLine blameLine = _blame.Lines[i];
-                GitBlameHeader blameHeader = _blame.FindHeaderForCommitGuid(blameLine.CommitGuid);
-                if (i > 0 && _blame.Lines[i - 1].CommitGuid == blameLine.CommitGuid)
+                if (line.Commit == lastCommit)
                 {
-                    blameCommitter.AppendLine(new string(' ', 200));
+                    gutter.Append(' ', 200).AppendLine();
                 }
                 else
                 {
-                    blameCommitter.AppendLine(
-                        (blameHeader.Author + " - " + blameHeader.AuthorTime + " - " + blameHeader.FileName +
-                         new string(' ', 100)).Trim('\r', '\n'));
+                    gutter.Append(line.Commit.Author);
+                    gutter.Append(" - ");
+                    gutter.Append(line.Commit.AuthorTime.ToString(CultureInfo.CurrentUICulture));
+                    gutter.Append(" - ");
+                    gutter.Append(line.Commit.FileName);
+                    gutter.Append(' ', 100).AppendLine();
```

Comment:
`Append(' ', 100)` is marginally better than `Append(new string(' ', 100))`.

Does anyone know why we need to append so many spaces to the end of _every_ line in the blame control gutter?

Reply:
I *think* this EOL white-space is so there's some text to highlight on mouse-over. The highlight is done via text background colour. If it could be done with a solid rectangle around the text, the extra spaces could be removed. 

I've added this as a comment to the code.